Update animals

erickma1 opened the pull request:
In this milestone, I did the following;

- Update animals table
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

- Update animals data and Rollback inside transaction

![1_update_digimon_pokemon](https://github.com/erickma1/vet_clinic/assets/110662611/63729032-ef2a-4aac-9a06-dc495c3cea92)

- Delete all data and rollback

![2_delete_select_rollback](https://github.com/erickma1/vet_clinic/assets/110662611/c62be0e5-01fa-4f1d-830f-94ce4d3acc4c)

- Delete and update animal table

![3_savepoint](https://github.com/erickma1/vet_clinic/assets/110662611/86779fbc-00f8-44a5-bc2e-1238b1b0458f)

- How many animals are there?

![4_count_total](https://github.com/erickma1/vet_clinic/assets/110662611/61abac5f-1fc6-40de-aa8a-b2509f83237e)

- How many animals are there?

![5_count_escape_less_1](https://github.com/erickma1/vet_clinic/assets/110662611/86012d61-65c4-464b-8423-0dd02cd82e65)

- What is the average weight of animals?

![6_avearge_weight](https://github.com/erickma1/vet_clinic/assets/110662611/82ea7e01-a399-4cfd-a2a0-5a3ae628da64)

- Who escapes the most, neutered or not neutered animals?

![7_neutured_sum](https://github.com/erickma1/vet_clinic/assets/110662611/ce57e1b7-879f-4266-a766-e251ff620bdb)

- What is the minimum and maximum weight of each type of animal?

![8_species_min_wei](https://github.com/erickma1/vet_clinic/assets/110662611/a126ec5f-804c-4f02-a48b-c22f5c2f870d)

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

![9_species](https://github.com/erickma1/vet_clinic/assets/110662611/ccb19db5-3f96-4c93-8293-b1f853607155)

erickma1 committed:
Update data.sql

erickma1 committed:
Update README.md

erickma1 committed:
Update queries.sql

erickma1 committed:
Update schema.sql

Johnadibe commented on data.sql:
```diff
@@ -7,3 +7,12 @@ VALUES
+
+INSERT INTO animals (name, date_of_birth, escape_attempts,neutered, weight_kg) VALUES 
+  ('Charmandor', '2020-02-08', 0, FALSE, -11),
+  ('Plantmon','2021-11-15', 2, TRUE, 5.7),
```
> - [ ] Hello @erickma1, According to the project requirements as stated [here](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Animal%3A%20Her%20name%20is%20Plantmon.%20She%20was%20born%20on%20Nov%2015th%2C%202021%2C%20and%20currently%20weighs%20%2D5.7kg.%20She%20is%20neutered%20and%20she%20has%20tried%20to%20escape%202%20times), the provided weight_kg data for Plantmon is `-5.7` and not `5.7`. Please kindly make this adjustment and you can update the data in your screenshots👍

Johnadibe commented on data.sql:
```diff
@@ -7,3 +7,12 @@ VALUES
    ('Devimon', '2017-05-12', 11, true, 5);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts,neutered, weight_kg) VALUES 
+  ('Charmandor', '2020-02-08', 0, FALSE, -11),
```
> - [ ] Hello @erickma1, I noticed that you spelled "Charmander" as "Charmandor" in the database. It's a small mistake, but it's important to be careful when dealing with databases, as even small errors can lead to problems. Kindly make the corrections.👍

Johnadibe reviewed (changes requested):
# STATUS: REQUIRED CHANGES ♻️
### Hi @erickma1👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![mario_ladders](https://user-images.githubusercontent.com/20563252/121369265-4e418880-c901-11eb-848c-1eff06d5a4cf.gif)

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Modified the schema.rb file to add a species column✔
✅ Modified data.rb file to add the provided data✔
✅ Modified the queries.rb file to perform transactions✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

erickma1 committed:
Update data.sql

erickma1 commented:
In this PR, I did the following;

- Changed the spelling of  "Charmandor" as "Charmander" in the database to match the project requirements
- Changed the provided weight_kg data for Plantmon to -5.7

The updates above can be verified from the screenshot below in the Update animals data and Rollback inside transaction query

![1_update_digimon_pokemon](https://github.com/erickma1/vet_clinic/assets/110662611/6bff57ff-0894-4b8c-addc-0f162630109c)

Johnadibe reviewed (approved):
# Status: Approved ✔️
### Hi @erickma1👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![LeonardoInGatsby](https://user-images.githubusercontent.com/38088818/233992044-ad00b20f-3f2f-4c86-8b94-1e613432af5f.gif)

## Highlights
- Implemented the changes requested by the previous reviewer✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
